django/core/serializers/jsonl.py

```diff
@@ -39,19 +39,30 @@ def getvalue(self):
         return super(PythonSerializer, self).getvalue()
 
 
-def Deserializer(stream_or_string, **options):
+class Deserializer(PythonDeserializer):
     """Deserialize a stream or string of JSON data."""
-    if isinstance(stream_or_string, bytes):
-        stream_or_string = stream_or_string.decode()
-    if isinstance(stream_or_string, (bytes, str)):
-        stream_or_string = stream_or_string.split("\n")
-
-    for line in stream_or_string:
-        if not line.strip():
-            continue
+
+    def __init__(self, stream_or_string, **options):
+        if isinstance(stream_or_string, bytes):
+            stream_or_string = stream_or_string.decode()
+        if isinstance(stream_or_string, str):
```

Comment:
```suggestion
        if isinstance(stream_or_string, (bytes, str)):
```
This was also ran for bytes previously, was this a conscious change? Might be worth adding a test for bytes input (there isn't one in the existing test suite) to check everything is ok or this makes sense
The pyyaml also is missing coverage for bytes input
Both of these are existing coverage issues but it would be nice to resolve them considering we refactoring this area

Reply:
I didn't quite get what you meant because that check for bytes was unnecessary. If there's a `bytes` object, it gets converted into `str` in its above `if`. Anyway, I have added tests for bytes input and the code works well.